scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/Readme.md

```diff
@@ -0,0 +1,172 @@
+
+``` yaml
+internal_package_name:
+  # parameter list (See below)
+```
+
+### Documentation and Naming Parameters
+- Size source to be added to footprint documentation field (`size_source`) {url}
+- Footprint library name (`library`) {string}
+- Footprint name generation control
+  - Device type added as prefix. Example QFN. (`device_type`) {string}
+  - Selction of different name format. Example for LGA footprints. (`use_name_format`) {enum(`LGA`, `not LGA`)}
+  - Manufacturer name. Will be added as a prefix if given. (`manufacturer`) {string}
+  - Part number. Will be added as a prefix if given. (`part_number`) {string}
+  - Suffix: A custom suffix (added after pin pitch in default naming format). Can include parameters `pad_x` and `pad_y`.
+  - Custom naming (`custom_name_format`) {python format string}
+    - The full default format string is `{man:s}_{mpn:s}_{pkg:s}-{pincount:d}-1EP_{size_x:g}x{size_y:g}mm_P{pitch:g}mm{suffix:s}_EP{ep_size_x:g}x{ep_size_y:g}mm_Mask{mask_size_x:g}x{mask_size_y:g}mm{suffix2:s}{vias:s}` (The same parameters can be used in your custom format. Exposed pad parameters are not available for components without exposed pad.)
+
+_Note: Contributions intended for the official library shall not include the manufacturer or part number unless the footprint is specific to that manufacturer or part. Similarly avoid custom naming for official library contributions unless required to achieve the requested name (Example TI-specific naming)._
+
+### Package Dimensions
+![dimension example](./documentation/dimension_system.svg)
+- Body size (`body_size_x`, `body_size_y`, `overall_height`) {dimension}
+- Lead dimensions:
+  - Lead width (`lead_width`) {dimension}
+  - Lead length (`lead_len`) {dimension}
+    - Optional for specifying different length for horizontal and vertical leads (`lead_len_H`, `lead_len_V`) {dimension}
+- Lead pitch, currently equal for all sides (`pitch`) {float}
+
+#### Pull Back Leads
+By default the leads are assumed to extend to the body edge. Some components however have the leads pulled back a bit. A typical example are packages that are called LGA by manufacturers.
+These packages need to select the pull back option for the IPC class (`ipc_class: 'qfn_pull_back'`) and need to be able to specify where the lead is.
+
+![dimension example for pull back leads](./documentation/dimension_system_pull_back.svg)
+
+- Most manufacturers give the lead pull back distance (`lead_to_edge`) {dimension}
+   - Alternatively the center position can be given (`lead_center_pos_x` and `lead_center_pos_y`) {dimension}
+   - Another alternative is to give the center to center dimenions for leads. (`lead_center_to_center_x` and `lead_center_to_center_y`) {dimension}
+- Some manufacturers give the body to inner edge dimension instead of the lead length (`body_to_inside_lead_edge`) {dimenions}
+
+### Pad Count
+- Pad count (`num_pins_x`, `num_pins_y`) {int}
+  - `num_pins_x`=0 is used for generating DNF like packages.
```

Comment:
'DNF' should be 'DFN' here and the line below.

Reply:
And add a hyphen after it, to make 'DFN-like'.

Reply:
Fixed also in gullwing readme (The hyphen note)